compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureExpressionVisitor.java

```diff
@@ -0,0 +1,783 @@
+/*
```

Comment:
I think, we could remove this class. I have initially, moved it separately because I thought we could reuse below functionality for both transaction and query desugar purposes.  But, we no longer need this for transactions. Isn't it ?

Reply:
But there is a usage in transaction desugar. i will check whether we need it or not